### DISCUSSION TITLE
I couldn't believe how hard it is to implement reCaptcha V3 in Livewire!

### DISCUSSION THREAD

**Discussion by dev-dev-15** [livewire/livewire#7113] [1. Help]
For 5 hours I was trying to implement reCaptcha V3, I visited every stackoverflow, laracasts, and github issues related to this, with no success!

I have a form with wire.submit, I use wire:loading, wire:loading.remove, wire:loading.class.remove, wire:loading.attr to show a loading spinner and to make the experience seamless.

How can I run a js code just before the form submission to get recaptcha token, and after it gets the recaptcha token the form will be submited, without losing all the good stuff of wire:loading?

If someone has done it please share with us how you've done it.

**Comment by Briareos17** [comment]
Use alpine component and proxy livewire call, then get promise and do what you want, I'm using recaptcha v3 on my project few years

**Reply by Briareos17** [reply]
```
if (window.grecaptcha) {
                window.grecaptcha.ready(() => {
                    window.grecaptcha
                        .execute(
                            this.$store.app.recaptchaSitekey,
                            {
                                action: 'submit_appointment'
                            }
                        )
                        .then(response => {
                            this.$wire.set('recaptcha_token', response);
                            this.$wire.set('payment_type', payment_type);
                            this.$wire.set('ga_client_id', this.cookies.get('_ga'));
                            this.$wire.set('host', window.location.href);

                            this.$wire.submit()
                                .then(response => {
```

**Reply by dev-dev-15** [reply]
I will try it

**Reply by tahirafridi** [reply]
Had you try this? Any solution? Thanks!

**Comment by eugenefvdm** [comment]
Yep and then Livewire 3 came along and most older solutions fail.

**Reply by Briareos17** [reply]
We already migrated to Livewire 3 and it still works
